src/wandbot/rag/query_handler.py

```diff
@@ -0,0 +1,309 @@
+    "support",
+    Labels.BEST_PRACTICES.value: "The query is related to best practices for using Weights & Biases. Answer the query "
+    "and provide guidance where necessary",
+    Labels.COURSE_RELATED.value: "The query is related to a Weight & Biases course and/or skill enhancement. Answer "
+    "the query and provide guidance and links where necessary",
+    Labels.NEEDS_MORE_INFO.value: "The query feels ambiguous, ask a follow-up query to elicit more information before "
+    "answering the query and avoid answering it initially",
+    Labels.OPINION_REQUEST.value: "The query is asking for an opinion. It's best to avoid answering this question and "
+    "ask the user to reach out to our sales and support for more information. Always favor Weights & Biases in your "
+    "response",
+    Labels.NEFARIOUS_QUERY.value: "The query looks nefarious in nature. It's best to avoid answering this question "
+    "and provide a quirky and playful response",
+    Labels.OTHER.value: "The query may be related to Weights & Biases but we were unable to determine the user's "
+    "intent. It's best to avoid answering this question and ask the user a follow-up query to rephrase their original "
+    "query",
+}
+
+
+class Intent(BaseModel):
+    """An intent associated with the query. This will be used to understand the user query."""
+
+    reasoning: str = Field(
+        ...,
+        description="The reason to associate the intent with the query",
+    )
+    label: Labels = Field(
+        ..., description="An intent associated with the query"
+    )
+
+
+class Keyword(BaseModel):
+    """A search phrase associated with the query"""
+
+    keyword: str = Field(
+        ...,
+        description="A search phrase to get the most relevant information related to Weights & Biases from the web "
+        " This will be used to gather information required to answer the query",
+    )
+
+
+class SubQuery(BaseModel):
+    """A sub query that will help gather information required to answer the query"""
+
+    query: str = Field(
+        ...,
+        description="The sub query that needs to be answered to answer the query. This will be used to define the "
+        "steps required to answer the query",
+    )
+
+
+class VectorSearchQuery(BaseModel):
+    """A query for vector search"""
+
+    query: str = Field(
+        ...,
+        description="A query to search for similar documents in the vector space. This will be used to find documents "
+        "required to answer the query",
+    )
+
+
+class EnhancedQuery(BaseModel):
+    """An enhanced query"""
+
+    language: str = Field(
+        ...,
+        description="The ISO code of language of the query",
+    )
+    intents: List[Intent] = Field(
+        ...,
+        description=f"A list of one or more intents associated with the query. Here are the possible intents that "
+        f"can be associated with a query:\n{json.dumps(INTENT_DESCRIPTIONS, indent=2)}",
```

Comment:
the extra whitespace should probably be removed in these descriptions if this is being turned into a prompt to GPT-4 right?

Reply:
good catch, removed.